.github/workflows/spellcheck.yml

```diff
@@ -0,0 +1,19 @@
+name: Spell Check
```

Comment:
I personally don't think it is required to make it a GitHub Action and make it always run on Kong's CI as a gatekeeper; incorrect "considered misspellings" will result in the CI pipeline status to **red 🔴**, and developers have to deal with `.ignore_words.txt` and `.ignore_lines.txt`, which is not supposed to be their responsibility (to fight with bad tooling).

Reply:
The reason why this tool works for Kong repo is that it has a ton of misspelled words in the dictionaries which can be confidently considered it's wrong. (compared with Vale that the docs repo is using)

And I personally don't think there was a good tooling that can be used without adding any "ignored/exclude" thing for the programming repo.

I really don't think adding a few ignored terms/lines costs any pains compared with the good side it brings to this repo.

Reply:
From my perspective, it makes sense to add a script under scripts/ folder to check spelling mistakes for convenience. However, it shouldn't be a blocker for any PR. 

**I am not convinced that I didn't see any popular open source project use such a tool running against every PR.**

Reply:
Mayo, do you think it is valuable to catch spelling mistakes at the PR stage and do so in an automated manner rather than relying on maintainers to check?

Reply:
@hbagdi It’s very common that a spellcheck tool reports a false-positive warning, then developers need to learn how to write .ignore rules. I don’t think it is that much valuable for identifying just spelling mistakes. More effort is needed to maintain the tools.

Reply:
It’s a manner that using a spellcheck tool for Markdown docs, our docs.kongkq.com also use Vale to check spelling errors. But I don’t think it’s necessary to enabe it for all code files.